lib/client/interfaces.go

```diff
@@ -403,16 +403,34 @@ func (k *Key) TeleportTLSCertificate() (*x509.Certificate, error) {
 	return tlsca.ParseCertificatePEM(k.TLSCert)
 }
 
-// KubeTLSCertificate returns the parsed x509 certificate for
-// authentication against a named kubernetes cluster.
-func (k *Key) KubeTLSCertificate(kubeClusterName string) (*x509.Certificate, error) {
+// KubeX509Cert returns the parsed x509 certificate for authentication against
+// a named kubernetes cluster.
+func (k *Key) KubeX509Cert(kubeClusterName string) (*x509.Certificate, error) {
 	tlsCert, ok := k.KubeTLSCerts[kubeClusterName]
 	if !ok {
 		return nil, trace.NotFound("TLS certificate for kubernetes cluster %q not found", kubeClusterName)
 	}
 	return tlsca.ParseCertificatePEM(tlsCert)
 }
 
+// KubeX509Cert returns the tls.Certificate for authentication against a named
+// kubernetes cluster.
+func (k *Key) KubeTLSCert(kubeClusterName string) (tls.Certificate, error) {
+	tlsCert, ok := k.KubeTLSCerts[kubeClusterName]
+	if !ok {
+		return tls.Certificate{}, trace.NotFound("TLS certificate for kubernetes cluster %q not found", kubeClusterName)
+	}
+	keyPem, err := k.PrivateKey.RSAPrivateKeyPEM()
+	if err != nil {
+		return tls.Certificate{}, trace.Wrap(err)
+	}
+	cert, err := keys.X509KeyPair(tlsCert, keyPem)
+	if err != nil {
+		return tls.Certificate{}, trace.Wrap(err)
+	}
+	return cert, nil
```

Comment:
Maybe realign naming a bit for more clarity? (We also can apply `certPEM` inside `KubeTLSCert()`)
```suggestion
	certPEM, ok := k.KubeTLSCerts[kubeClusterName]
	if !ok {
		return tls.Certificate{}, trace.NotFound("TLS certificate for kubernetes cluster %q not found", kubeClusterName)
	}
	keyPem, err := k.PrivateKey.RSAPrivateKeyPEM()
	if err != nil {
		return tls.Certificate{}, trace.Wrap(err)
	}
	tlsCert, err := keys.X509KeyPair(certPEM, keyPem)
	if err != nil {
		return tls.Certificate{}, trace.Wrap(err)
	}
	return tlsCert, nil
```